### PR TITLE
Log n audit

### PR DESCRIPTION
**Pull request by sxarism**

### CODE REVIEW

File: services/TS29222_CAPIF_Auditing_API/logs/__main__.py [hunk 1]
```diff
@@ -4,16 +4,29 @@
...
-                pythonic_params=True)
 
-    app.run(port=8080)
+app = connexion.App(__name__, specification_dir='./openapi/')
+app.app.json_encoder = encoder.JSONEncoder
+app.add_api('openapi.yaml',
+            arguments={'title': 'CAPIF_Auditing_API'},
+            pythonic_params=True)
+
+app.app.config["JWT_SECRET_KEY"] = "this-is-secret-key"
```
Comment (raxonros) [inline services/TS29222_CAPIF_Auditing_API/logs/__main__.py]: You can remove all lines related with JWT and mosquitto and Matt

File: services/TS29222_CAPIF_Auditing_API/logs/controllers/default_controller.py [hunk 2]
```diff
@@ -37,22 +51,30 @@ def api_invocation_logs_get(aef_id=None, api_invoker_id=None, time_range_start=N
...
     :param supported_features: To filter irrelevant responses related to unsupported features
     :type supported_features: str
 
     :rtype: InvocationLog
     """
+
+    cert_tmp = request.headers['X-Ssl-Client-Cert']
+    cert_raw = cert_tmp.replace('\t', '')
+
+    cert = x509.load_pem_x509_certificate(str.encode(cert_raw), default_backend())
```
Comment (raxonros) [inline services/TS29222_CAPIF_Auditing_API/logs/controllers/default_controller.py]: you can remove this code.
We are talking how to implement the identity verification in nginx using the DN in certificates

File: services/TS29222_CAPIF_Auditing_API/logs/core/auditoperations.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+from ..db.db import MongoDatabse
+from ..encoder import JSONEncoder
+from bson import json_util
+from ..models.protocol import Protocol
+
+
+class AuditOperations:
+
+    def __init__(self):
```
Comment (raxonros) [inline services/TS29222_CAPIF_Auditing_API/logs/core/auditoperations.py]: The connection to the database is not made from the class but from an abstract class and we make the class that performs the operations on the database inherit from it (Review code)

File: services/TS29222_CAPIF_Auditing_API/logs/core/auditoperations.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        self.db = MongoDatabse()
+        self.mimetype = 'application/json'
+
+    def get_logs(self, aef_id, api_invoker_id, time_range_start, time_range_end, api_id, api_name, api_version, protocol, operation, result, resource_name, src_interface, dest_interface, supported_features):
+
+        mycol = self.db.get_col_by_name(self.db.invocation_logs)
+        users_col = self.db.get_col_by_name(self.db.capif_users)
+
+        try:
```
Comment (raxonros) [inline services/TS29222_CAPIF_Auditing_API/logs/core/auditoperations.py]: try to add logs, review main branch code to add and configure logger

File: services/TS29222_CAPIF_Auditing_API/logs/core/check_user.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import sys
+from ..db.db import MongoDatabse
+
+class CapifUsersOperations:
+
+    def __init__(self):
```
Comment (raxonros) [inline services/TS29222_CAPIF_Auditing_API/logs/core/check_user.py]: You can remove this file

File: services/TS29222_CAPIF_Logging_API_Invocation_API/api_invocation_logs/__main__.py [hunk 6]
```diff
@@ -4,16 +4,29 @@
...
-                pythonic_params=True)
 
-    app.run(port=8080)
+app = connexion.App(__name__, specification_dir='./openapi/')
+app.app.json_encoder = encoder.JSONEncoder
+app.add_api('openapi.yaml',
+            arguments={'title': 'CAPIF_Logging_API_Invocation_API'},
+            pythonic_params=True)
+
+app.app.config["JWT_SECRET_KEY"] = "this-is-secret-key"
```
Comment (raxonros) [inline services/TS29222_CAPIF_Logging_API_Invocation_API/api_invocation_logs/__main__.py]: You can remove jet, mosquitto and Matt references

File: services/TS29222_CAPIF_Logging_API_Invocation_API/api_invocation_logs/controllers/default_controller.py [hunk 7]
```diff
@@ -18,6 +28,24 @@ def aef_id_logs_post(aef_id, invocation_log):  # noqa: E501
 
     :rtype: InvocationLog
     """
+
+    cert_tmp = request.headers['X-Ssl-Client-Cert']
```
Comment (raxonros) [inline services/TS29222_CAPIF_Logging_API_Invocation_API/api_invocation_logs/controllers/default_controller.py]: you can remove this code (identity verify)

File: services/TS29222_CAPIF_Logging_API_Invocation_API/api_invocation_logs/core/check_user.py [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import sys
+from ..db.db import MongoDatabse
+
+class CapifUsersOperations:
+
+    def __init__(self):
+        self.db = MongoDatabse()
```
Comment (raxonros) [inline services/TS29222_CAPIF_Logging_API_Invocation_API/api_invocation_logs/core/check_user.py]: You can remove this file

File: services/TS29222_CAPIF_Logging_API_Invocation_API/api_invocation_logs/core/invocationlogs.py [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+from ..db.db import MongoDatabse
+from ..encoder import JSONEncoder
+from ..models.problem_details import ProblemDetails
+
+
+class LoggingInvocationOperations:
+
+    def __init__(self):
+        self.db = MongoDatabse()
```
Comment (raxonros) [inline services/TS29222_CAPIF_Logging_API_Invocation_API/api_invocation_logs/core/invocationlogs.py]: The connection to the database is not made from the class but from an abstract class and we make the class that performs the operations on the database inherit from it (Review code)

File: services/TS29222_CAPIF_Logging_API_Invocation_API/config.yaml [hunk 10]
```diff
@@ -0,0 +1,12 @@
+mongo: {
+  'user': 'root',
+  'password': 'example',
+  'db': 'capif',
+  'col': 'invocationlogs',
```
Comment (raxonros) [inline services/TS29222_CAPIF_Logging_API_Invocation_API/config.yaml]: change the names col1, col2 etc to something more descriptive, invoker_col, log_col, etc

**Review by raxonros — COMMENTED** [review]
Revise comments

**Review by raxonros — COMMENTED** [review]
LGTM!

File: pac/Jenkins-sonar.groovy [hunk 11]
```diff
@@ -35,7 +35,7 @@ pipeline {
                     sh '''
                     mkdir $NETAPP_NAME
                     cd $NETAPP_NAME
-                    git clone --single-branch --branch $CHANGE_BRANCH $GIT_CAPIF_URL .
+                    git clone --single-branch --branch $CHANGE_BRANCH https://github.com/EVOLVED-5G/CAPIF_API_Services .
```
Comment (jorgems101) [inline pac/Jenkins-sonar.groovy:38]: GIT_CAPIF_URL variable must be used,no?

File: pac/Jenkins-sonar.groovy [hunk 12]
```diff
@@ -66,6 +66,9 @@ pipeline {
                                 if(array=="mosquitto"){
                                     return
                                 }
+                                if(array=="mosquitto"){
```
Comment (jorgems101) [inline pac/Jenkins-sonar.groovy:69]: this code block is repeated

**Review by alemosan1 — DISMISSED** [review]
LGTM

**Review by raxonros — APPROVED** [review]
LGTM

**Pull request merged**